.github/workflows/build_and_deploy.yml

```diff
@@ -0,0 +1,104 @@
+      # It is actually checking out the most recent version on this branch
+      - name: Checkout
+        uses: actions/checkout@master
+
+      # Visual Studio 2019 (windows-latest) doesn't include these files for some reason??
+      # Even stranger, it really only affects Python 3.5, but we run it across all versions for consistency
+      # https://social.msdn.microsoft.com/Forums/vstudio/en-US/06675e6d-9a0b-46f5-8de0-10237fab1ece/link-error-lnk1158?forum=vcgeneral
+      - name: Fix Windows
+        if: matrix.os == 'windows-latest'
+        run: |
+          Copy-Item "C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x86\rc.exe" -Destination "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN"
+          Copy-Item "C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x86\rcdll.dll" -Destination "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN"
+        shell: powershell
+
+      - name: Setting up Python
+        uses: actions/setup-python@v1.1.1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: ${{ matrix.architecture }}
+
+      - name: Updating pip
+        run: python -m pip install --upgrade pip
+        shell: bash
+
+      - name: Installing requirements
+        run: |
+          pip install -r requirements.txt
+          pip install -r build_tools/build_requirements.txt
+        shell: bash
+
+      - name: Building wheel
+        run: make bdist_wheel
+        shell: bash
+
+      - name: Installing generated wheel
+        run: pip install --pre --no-index --find-links dist/ pmdarima
+        shell: bash
+
+      - name: Running unit tests
+        run: |
+          if [ "${{ matrix.os }}" == "macos-latest" ]; then
+            export PMD_MPL_BACKEND=TkAgg
+          fi
+          pytest --showlocals --durations=20 --mpl --mpl-baseline-path=etc/pytest_images --pyargs pmdarima --benchmark-skip
+        shell: bash
+
+      # We build the source archive separately because of this: https://github.com/alkaline-ml/pmdarima/pull/136#discussion_r279781731
+      # This also implicitly runs `make version`, which -- if tagged -- gets the right version
+      - name: Building source distribution
+        run: make sdist
+        shell: bash
+
+      - name: Checking README compatibility
+        run: |
+          if python -c "from twine.commands.check import check; check(['dist/*'])" | grep  "warning"; then
+              echo "README will not render properly on PyPI"
+              exit 1
+          else
+              echo "README rendered appropriately"
+          fi
+        shell: bash
+
+      - name: Ensuring sdist can be installed
+        run: pip install dist/$(ls dist | grep tar)
+        shell: bash
+
+      # Use this one for testing: python -m twine upload --repository-url https://test.pypi.org/legacy/ --skip-existing dist/pmdarima-*
+      - name: Deploying to PyPI
+        if: github.event_name == 'tag' && success()  # Only deploy on tags
```

Comment:
Should we also add `&& [ -f pmdarima/VERSION ]` (or whatever equivalent) so we guarantee we only release if the VERSION file was created?

Reply:
Added in b39e1b73be3b2709da4527be5eb4d59e24abbf0d. We have to do a check in a previous step, then store output. I confirmed it [here](https://github.com/alkaline-ml/pmdarima/pull/263/checks?check_run_id=345830965#step:14:1)